src/sentry/scripts/similarity/index.lua

```diff
@@ -278,507 +370,334 @@ local function scale_to_total(values)
-    --[[
-    Calculate the similarity between an item's frequency and an array-like
-    table of candidate frequencies. This returns a table of candidate keys to
-    and [0, 1] scores where 0 is totally dissimilar and 1 is exactly similar.
-    ]]--
     local results = {}
-    for key, value in pairs(candidate_frequencies) do
-        table.insert(
-            results,
-            {
-                key,
-                table.ireduce(  -- sum, then avg
-                    table.imap(  -- calculate similarity
-                        table.izip(
-                            item_frequencies,
-                            value
-                        ),
-                        function (v)
-                            -- We calculate the "similarity" between two items
-                            -- by comparing how often their contents exist in
-                            -- the same buckets for a band.
-                            local dist = get_manhattan_distance(
-                                scale_to_total(v[1]),
-                                scale_to_total(v[2])
-                            )
-                            -- Since this is a measure of similarity (and not
-                            -- distance) we normalize the result to [0, 1]
-                            -- scale.
-                            return 1 - (dist / 2)
-                        end
-                    ),
-                    function (total, item)
-                        return total + item
-                    end,
-                    0
-                ) / configuration.bands
-            }
-        )
+    for candidate, bands in pairs(candidates) do
+        results[candidate] = table.count(bands)
     end
     return results
 end
 
-local function fetch_similar(configuration, time_series, index, item_frequencies)
-    --[[
-    Fetch the items that are similar to an item's frequencies (as returned by
-    `fetch_bucket_frequencies`), returning a table of similar items keyed by
-    the candidate key where the value is on a [0, 1] similarity scale.
-    ]]--
-    local candidates = fetch_candidates(configuration, time_series, index, item_frequencies)
-    local candidate_frequencies = {}
-    for candidate_key, _ in pairs(candidates) do
-        candidate_frequencies[candidate_key] = fetch_bucket_frequencies(
-            configuration,
-            time_series,
-            index,
-            candidate_key
+local function search(configuration, parameters, limit)
+    local possible_candidates = {}
+    local create_table = function ()
+        return {}
+    end
+
+    for i, p in ipairs(parameters) do
+        for candidate, hits in pairs(fetch_candidates(configuration, p.index, p.frequencies)) do
+            if hits >= p.threshold then
+                table.get_or_set_default(possible_candidates, candidate, create_table)[i] = hits
+            end
+        end
+    end
+
+    local candidates = {}
```

Comment:
I don't love this transformation from `possible_candidates` to `candidates` and will likely try and think of a cleaner way to this. This also just leaves `possible_candidates` hanging around after it's no longer needed